direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorageTest.java

```diff
@@ -631,34 +631,93 @@ public boolean onError(Throwable error) {
     assertEquals(3, observeHandle.getCurrentOffsets().size());
   }
 
-  @Test(expected = UnsupportedOperationException.class)
+  @Test
+  public void testObserveSinglePartitionOutOfMultiplePartitions() throws InterruptedException {
+    final int numPartitions = 3;
+    final InMemStorage storage = new InMemStorage();
+    final DataAccessor accessor =
+        storage.createAccessor(
+            direct, createFamilyDescriptor(URI.create("inmem:///test"), numPartitions));
+    final CommitLogReader reader = Optionals.get(accessor.getCommitLogReader(direct.getContext()));
+    final AttributeWriterBase writer = Optionals.get(accessor.getWriter(direct.getContext()));
+    final int numElements = 999;
+    final ConcurrentMap<Partition, Long> partitionHistogram = new ConcurrentHashMap<>();
+    // Elements are uniformly distributed between partitions.
+    final CountDownLatch elementsReceived = new CountDownLatch(numElements / numPartitions);
+    // Start observer.
+    final ObserveHandle observeHandle =
+        reader.observePartitions(
+            reader.getPartitions().subList(0, 1),
+            new LogObserver() {
+
+              @Override
+              public void onRepartition(OnRepartitionContext context) {
+                assertEquals(numPartitions, context.partitions().size());
+              }
+
+              @Override
+              public boolean onNext(StreamElement ingest, OnNextContext context) {
+                partitionHistogram.merge(context.getPartition(), 1L, Long::sum);
+                context.confirm();
+                elementsReceived.countDown();
+                return elementsReceived.getCount() > 0;
```

Comment:
Maybe here, we can read the key and check that it satisfies some condition, like `Integer.parseInt(ingest.getKey().split("_")) % 3 == 0`